### ISSUE TITLE
Feature Request: Mass Transfer to also make changes to CloudFlare DNS record for Hacker CMS and WordPress?

### ISSUE THREAD

**Issue by xet7** [sandstorm-io/sandstorm#3163]
Hi,
Mass Transfer code most likely knows behind scenes:
- published Hacker CMS and WordPress URLs for source and destination
- *.sandcats.io URLs.

Based on this, would it be possible for mass transfer tool to generate some of these?
- log file, that contains source and destination published Hacker CMS and WordPress URLs
- script that makes changes to all CloudFlare DNS records

Or maybe:
- Have Mass control panel that does all of moving grains and CloudFlare DNS changes, so when grain is moved, DNS records are updated directly?

**Comment by ocdtrekkie** [comment]
Sandstorm wouldn't have your Cloudflare API key though. And many grains that support web publishing know nothing about the URL being directed towards them, so such a feature would be very specifically just usable for WordPress right now, pretty much.

I think there'd be a particularly interesting point, where if Cloudflare is workable for Sandstorm in general (wildcard DNS being affordable, which the CEO of Cloudflare sounded at least amicable to), where a feature could be built allowing Sandstorm to configure and manage your Cloudflare setup for it.

However, what would probably be easier within the realm of what Sandstorm currently does, is it could probably report what grains you transfer which have things like web publishing enabled, so you know what links you need to re-establish.

**Comment by xet7** [comment]
@ocdtrekkie 

I don't know is there any API to get published grain URL info from self-hosted *.sandcats.io server. Would that info be somewhere inside WordPress grain or Hacker CMS grain, because it's visible there. Probably with some code changes such API could be exposed with some API key?

Then some code running at my laptop could have both sandcats server and CloudFlare keys and do the changes.

I'm just thinking what would be most user friendly way for this.

**Comment by xet7** [comment]
At CloudFlare workers, there is limited space for KV store. Maybe some data could be saved to grains, like in SandForms? And loaded from there.

I'm just thinking, grains are still very good technology. Just today I did get email about security problem concerning Docker.

**Comment by ocdtrekkie** [comment]
I mean, WordPress you have to give your public publishing URL so it can make valid links. But other apps don't do that.

Cloudron manages the DNS for all the apps you have installed, but it does that holistically from the start. Whereas say blog.jacobweisz.com actually points to a grain on Oasis, but my Oasis grain doesn't actually know that.

**Comment by xet7** [comment]
@ocdtrekkie 

Probably you will also move your Oasis grains elsewhere:
https://sandstorm.io/news/2019-09-15-shutting-down-oasis

**Comment by ocdtrekkie** [comment]
Of course. I'm going to be seeing if I can set Sandstorm up on-prem at home.

I'm just thinking right now Sandstorm doesn't manage your DNS setup at all. And mass transfers probably isn't the right place to start doing that all of a sudden. But there's merit in the idea of Sandstorm managing your DNS from the get-go someday.
